.github/workflows/ci.yml

```diff
@@ -0,0 +1,156 @@
+name: CI
+on:
+  push:
+    branch:
+      - master
+      - develop
+      - release**
+    tags:
+      - v**
+
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.5'
+          - 'nightly'
+        os:
+          - ubuntu-latest
+        arch:
+          - x64
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup julia
+        uses: julia-actions/setup-julia@v1
+        with:
+          version: ${{ matrix.version }}
+          arch: ${{ matrix.arch }}
+      
+      - name: Cache Artifacts
+        uses: actions/cache@v1
+        env:
+          cache-name: cache-artifacts
+        with:
+          path: ~/.julia/artifacts
+          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/Project.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ env.cache-name }}-
+            ${{ runner.os }}-test-
+            ${{ runner.os }}-
+
+      - name: Julia Build Pkg
+        uses: julia-actions/julia-buildpkg@latest
+      
+      - name: Git Test Credentials
+        run: |
+          git config --global user.name Tester
+          git config --global user.email te@st.er
+      
+      - name: Run tests
+        uses: julia-actions/julia-runtest@latest
+        env:
+          IIF_TEST: false
+          SKIP_CGDFG_TESTS: true
+        continue-on-error: ${{ matrix.version == 'nightly' }}
+      
+      - name: Process Coverage
+        uses: julia-actions/julia-processcoverage@v1
+        
+      - name: Code Coverage
+        uses: codecov/codecov-action@v1
+        with:
+          file: lcov.info
+        # run: |
+        #   julia -e 'using Pkg; cd(Pkg.dir("DistributedFactorGraphs")); Pkg.add("Coverage"); using Coverage; Codecov.submit(process_folder())'
+
+  test-cfg:
+    if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/develop' || github.ref == 'refs/heads/release**'
+    name: Documentation
+    runs-on: ubuntu-latest
+    steps:
+          - name: Checkout
+            uses: actions/checkout@v2
+
+          - name: Setup julia
+            uses: julia-actions/setup-julia@v1
+            with:
+              version: 1.4
+              arch: x64
+
+          - name: Install neo4j
```

Comment:
this was giving issues on Travis, will be cool if it just works again here on Actions